### PR TITLE
Feature/clean code

### PR DESCRIPTION
**Pull request by HFG43**
Hi there! We are sending this pull request for you to review our code. Thanks in advance for your feedback!
@francksefu & @HFG43 

Milestones 🎯:
📌 Work on Pair Programming
📌 Ensure DRY, YAGNI, KISS, and SOLID principles are applied
📌 Ensure `main.rb` is only for starting the code and asking for initial choice
📌 Test the app in the command line

🎖️ General requirements:
Gitflow
No linter errors
Ensure the use of best practices and Ruby style

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -13,17 +14,6 @@ def options_menu
...
-  when 2 then app.list_all_people
-  when 3 then app.create_person
-  when 4 then app.create_a_book
-  when 5 then app.create_a_rental
-  when 6 then app.rental_person_id
-  end
-end
-
 def main
   app = App.new
```
Comment (V-Blaze) [inline main.rb:19]: - You have done great so far but there are a few changes you need to make before moving to the next milestone. :smile: :+1: 


- [ ] Follwoing the [SOLID](`https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid`) PRINCIPLE AND DRY. You need to refactore your `create_person` and also the `create_student and create_teacher` method. All requests for input and input validation should've been placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input.
-  Also the input for `Age, name` is repeated in create person and teacher which doesn't follow the dry principle. I'll suggest you take note of these and update your code. :smile: :wave:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @HFG43, @francksefu  

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

almost-there


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- App class extracted to its own file✔
-  Main.rb only takes user inputs👋 
-  Correct Git flow✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.rb (hunk 1 above)
Comment (HFG43) [reply main.rb:19]: @V-Blaze thanks for your feedback! We will review these required changes!

File: app.rb [hunk 2]
```diff
@@ -59,17 +60,11 @@ def create_a_student
     puts "\n"
   end
 
-  def create_a_teacher
+  def create_a_teacher(name, age)
     type = 'Teacher'
     puts 'Specialization: '
     specialization = gets.chomp
```
Comment (ridwanediallo) [inline app.rb:66]: - [ ] Great done on getting rid of the redundant code 👏. Please make sure that your methods in `app.rb,` such as `create person,` `create student,` and `create teacher,` as he pointed out, are only doing one thing, though, per the prior reviewer's request. Hence, in order to accomplish that, you should ensure that anything that receives user inputs is placed in `main.rb` or inside other classes or methods. 👌

### Hint: :point_down: 
Every `gets.chomp` should be moved


- [ **OPTIONAL**]  Please do the same thing with the other remaining methods like: `create_a_book`, `create_a_rental`, and `rental_person_id`. 😊

**Review by ridwanediallo — CHANGES_REQUESTED** [review]
# Changes Required ♻️
## Hi team, 


AlmostThereReachGIF

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- You have have successfully removed the duplicate code. ✔


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 2 above)
Comment (HFG43) [reply app.rb:66]: @ridwanediallo Thanks for the hint! It helped understanding

**Review by mikethreels — APPROVED** [review]
Hi team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- good job following the instructions of the previous CR.
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**